subprojects/messaging/src/main/java/org/gradle/internal/remote/internal/hub/MessageHubBackedObjectConnection.java

```diff
@@ -85,17 +97,19 @@ public void useParameterSerializers(SerializerRegistry serializer) {
     }
 
     public void connect() {
-        if (methodParamClassLoader == null) {
+        ClassLoader methodParamClassLoader;
+        if (methodParamClassLoaders.size() == 0) {
             methodParamClassLoader = getClass().getClassLoader();
+        } else {
+            methodParamClassLoader = new CachingClassLoader(new MultiParentClassLoader(methodParamClassLoaders));
```

Comment:
Perhaps also deal with the `size() == 1` case, as using `MultiParentClassLoader` is quite expensive. We also shouldn't need more than 1 ClassLoader at this point, as the plugins ClassLoader can see the log event types, just register the incoming log event handler last instead of first.